### PR TITLE
Merging view : refactoring Network.BusView & Network.BusBreakerView

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Try to fit IIDM API for Network.BusView & Network.BusBreakerView

**What is the current behavior?** *(You can also link to an open issue here)*
Network.BusView & Network.BusBreakerView adapters are not nested classes of the relevant views

**What is the new behavior (if this is a feature change)?**
Network.BusView & Network.BusBreakerView adapters are nested classes of the relevant views

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java [hunk 1]
```diff
@@ -37,17 +37,116 @@
     /** To listen events from merging network */
     private final NetworkListener listener = new MergingNetworkListener();
 
-    private final NetworkBusBreakerViewAdapter busBreakerView;
+    static class BusBreakerViewAdapter implements Network.BusBreakerView {
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: Why this class is private package? Private could be enough?
It seems it's required by unit tests. It's usually a bad idea to make the model weaker just for unit test

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java [hunk 2]
```diff
@@ -37,17 +37,116 @@
...
+
+        @Override
+        public int getSwitchCount() {
+            return (int) getSwitchStream().count();
+        }
+    }
+
+    private final BusBreakerViewAdapter busBreakerView;
+
+    static class BusViewAdapter implements Network.BusView {
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: Same question here

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java (hunk 1 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: My mistake, i forgot to put private after to move BusBreakerViewAdapter class into MergingView class

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java (hunk 2 above)
Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: Same answer

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java [hunk 3]
```diff
@@ -37,17 +37,116 @@
...
+        public Bus getBus(final String id) {
+            final List<Network.BusView> views = index.getNetworkStream()
+                    .map(Network::getBusView).collect(Collectors.toList());
+            for (final Network.BusView bv : views) {
+                final Bus bus = bv.getBus(id);
+                if (Objects.nonNull(bus)) {
+                    return index.getBus(bus);
+                }
+            }
+            return null;
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: Should be implemented using lambdas:
```
return index.getNetworkStream()
        .map(n -> n.getBusView().getBus(id))
        .findFirst()
        .map(index::getBus)
        .orElse(null);
```
Not totally sure of my proposal, so I let you test it

Comment (tadam50) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java]: It works ... 
I just add a filter on nullalbe bus

**Pull request merged**